### ISSUE TITLE
Issue in json_setf

### ISSUE THREAD

**Issue by wahab80** [cesanta/frozen#20]
Hi,

I want to manipulate json data for example 
	"{ "a": 123, "b": [ 1 ], "c": true }"   // this is taken from your unit_test.c file

Want to change "a" value to 7
	json_setf(szJSON, strlen(szJSON), &out, ".a", "%d", 7);

I am getting following output
	{ "a": 7, "b": [ 1 ], "c": t

Last 5 characters get truncated.

Could you please look into this issue.
Note: above string works if we remove SPACE after COMMA and SPACE inside the ARRAY

Here is my code:

void main()
{
	char *szJSON = "{ \"a\": 123, \"b\": [ 1 ], \"c\": true }";
	char buf[1024];
	struct json_out out = JSON_OUT_BUF(buf, sizeof(buf));

	
	memset(buf, 0, 1024);
	json_setf(szJSON, strlen(szJSON), &out, ".a", "%d", 7);
	printf("%s\n\n", buf);

	getch();
}

**Comment by JonathanWitthoeft** [comment]
Perhaps the problem is you are not allocating memory for your char *szJSON. Try this and see if you get better results:

void main()
{
char szJSON[30] 
sprintf(szJSON, "{\"a\":123,\"b\":[1],\"c\":true}" );
char buf[1024];
struct json_out out = JSON_OUT_BUF(buf, sizeof(buf));

json_setf(szJSON, strlen(szJSON), &out, ".a", "%d", 7);
printf("%s\n\n", buf);

getch();
}

**Comment by cpq** [comment]
@wahab80 you've taken that from the unit_test that actually passes.

Your code is wrong - you must escape double quotes:

```
char *szJSON = "{ "a": 123, "b": [ 1 ], "c": true }";
```

**Comment by wahab80** [comment]
Sorry!  I have pasted wrong code. My code does contain escape for double qoutes and still have the issues.

After debugging the code now I know the reason for this issue. It is already mentioned in TODO list of "json_vprintf" that "vsnprintf implementation returns -1 on overflow rather needed size" for windows OS.

For now I have just increased my buf array size in json_vprintf to solve this issue.

Thanks for replying.
